### PR TITLE
Change the semantics of leaderWorkerSet Replicas

### PR DESCRIPTION
**Pull request by kerthcet**
#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes

Fixes https://github.com/kubernetes-sigs/lws/issues/96#issuecomment-2063027769

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/controllers/leaderworkerset_controller.go [hunk 1]
```diff
@@ -354,8 +354,7 @@ func (r *LeaderWorkerSetReconciler) updateConditions(ctx context.Context, lws *l
 	}
 
 	updateStatus := false
-	readyCount := 0
-	updatedCount := 0
+	readyCount, updatedCount, availableCount := 0, 0, 0
```
Comment (ahg-g) [inline pkg/controllers/leaderworkerset_controller.go:357]:
```suggestion
	readyCount, updatedCount, updatedAndReadyCount := 0, 0, 0
```

**Review by ahg-g — COMMENTED** [review]
Thanks! can we please update the documentation at https://github.com/kubernetes-sigs/lws/blob/main/api/leaderworkerset/v1/leaderworkerset_types.go#L179-L185?

```
	// Replicas tracks the total number of groups currently created (updated or not, ready or not).
	Replicas int32 `json:"replicas,omitempty"`

	// ReadyReplicas tracks the number of groups that are in ready state (updated or not).
	ReadyReplicas int32 `json:"readyReplicas,omitempty"`


	// UpdatedReplicas tracks the number of groups that have been updated (ready or not).
	UpdatedReplicas int32 `json:"updatedReplicas,omitempty"`

```

**Comment by liurupeng** [comment]
Thanks, looks good overall, I will continue the 0.2.0 release after this PR is merged.

File: pkg/controllers/leaderworkerset_controller.go (hunk 1 above)
Comment (kerthcet) [reply pkg/controllers/leaderworkerset_controller.go]: Done

**Review by kerthcet — COMMENTED** [review]
Done, PTAL

**Pull request merged**